rewriter/src/main/java/com/facebook/coresql/rewriter/OrderByRewriter.java

```diff
@@ -35,31 +36,38 @@
 public class OrderByRewriter
         extends Rewriter
 {
-    PatternMatcher<Set<AstNode>> matcher;
     public Set<AstNode> patternMatchedNodes;
     private static final String REWRITE_NAME = "ORDER BY without LIMIT";
 
     public OrderByRewriter()
     {
-        this.matcher = new OrderByPatternMatcher();
         this.patternMatchedNodes = new HashSet<>();
     }
 
     @Override
     public boolean rewritePatternIsPresent(String sql)
     {
         AstNode root = requireNonNull(parseStatement(sql));
-        patternMatchedNodes = matcher.matchPattern(root);
+        patternMatchedNodes = new OrderByPatternMatcher().matchPattern(root);
+        return !patternMatchedNodes.isEmpty();
+    }
+
+    private boolean rewritePatternIsPresent(AstNode root)
```

Comment:
why not directly call the public method? And if you really want to introduce a private method, at least modify the public method to use it as well?

Reply:
Combined the methods into one public method.

Done.